src/libraries/System.Runtime/tests/System.Globalization.Tests/CultureInfo/CultureInfoNames.cs

```diff
@@ -10,19 +10,31 @@ namespace System.Globalization.Tests
 {
     public class CultureInfoNames
     {
-        private static bool SupportFullIcuResources => (PlatformDetection.IsNotMobile && PlatformDetection.IsIcuGlobalization) || PlatformDetection.IsHybridGlobalizationOnApplePlatform;
+        private static bool SupportFullIcuResources =>
+            (PlatformDetection.IsNotMobile && PlatformDetection.IsIcuGlobalization) ||
+            PlatformDetection.IsHybridGlobalizationOnApplePlatform ||
+            PlatformDetection.IsBrowser;
+        
+        public static IEnumerable<object[]> SupportedCultures_TestData()
+        {
+            // Browser does not support all ICU locales but it uses JS to get the correct native name
+            if (!PlatformDetection.IsBrowser)
```

Comment:
Browser does not support full ICU, just 200 locales, so adding it to `SupportFullIcuResources` would be misleading. Hybrid mode supports only the locales supported by WASM, so limited amount, because HybridGlobalization in WASM does not cover all APIs. If we allowed full locales range, they would be working only partially.